src/restLevelClient/generateParameterTypes.ts

```diff
@@ -82,6 +108,67 @@ export function generateParameterInterfaces(
   ]);
 }
 
+function getRequestHeaderInterfaceDefinition(
+  parameters: Parameter[],
+  baseName: string
+): undefined | InterfaceDeclarationStructure {
+  // Check if there are any required headers
+  const headerParameters = parameters.filter(p => p.protocol.http?.in === "header");
+  if (!headerParameters.length) {
+    return undefined;
+  }
+  const headersInterfaceName = `${baseName}Headers`;
+  return {
+    kind: StructureKind.Interface,
+    isExported: true,
+    name: headersInterfaceName,
+    properties: headerParameters.map((h: Parameter) => {
+      const description = getLanguageMetadata(h.language).description;
+      return {
+        name: `"${getLanguageMetadata(h.language).name}"`,
+        ...(description && { docs: [{ description }] }),
+        type: primitiveSchemaToType(h.schema),
+        hasQuestionToken: true
```

Comment:
Is there any way to mark a header as required? If so we may want to calculate hasQuestionToken instead of hard coding it